Modules/_functoolsmodule.c

```diff
@@ -39,9 +55,16 @@ partial_new(PyTypeObject *type, PyObject *args, PyObject *kw)
         return NULL;
     }
 
+    PyObject *module = _PyType_GetModuleByDef(type, &_functools_module);
+    if (module == NULL) {
+        return NULL;
+    }
+    _functools_state *state = get_functools_state(module);
```

Comment:
Would you mind moving this into a helper method or macro?

Reply:
Christian , thanks for your comment. You are right. Using a common function or macro would be better. I will update it :)

Reply:
done.